core/src/calleeFunctions/helpers/uniswapAutoRouter.ts

```diff
@@ -55,19 +61,34 @@ export const fetchAutoRouteInformation = async function (
     walletAddress?: string
 ) {
     try {
-        const token = await getUniswapTokenBySymbol(network, collateralSymbol);
         const autoRouteData = await getUniswapAutoRoute(network, collateralSymbol, inputAmount, walletAddress);
-        const routes = autoRouteData.route[0].tokenPath.map(p => p.symbol);
-
+        const bestRoute = autoRouteData.route[0];
+        const route = bestRoute.tokenPath.map(p => {
+            if (!p.symbol) {
+                throw new Error(`Could not get symbol for token "${p.address}".`);
+            }
+            return p.symbol;
+        });
+        if (bestRoute.route.protocol !== Protocol.V3) {
+            throw new Error('Only V3 routes are supported.');
```

Comment:
apparently not always the router returns fees. So i check the protocol version to ensure that the fees are there.

Reply:
Are you talking about exchange fees? Is it a type problem or did you encounter such situations? (given line 58 above)

Reply:
yes, the fees for exchange are not always provided. It is a type problem. I have to ensure that the protocol version is correct - otherwise i cannot be sure that the fees are in the return object.